lib/Analysis/Allocation.cpp

```diff
@@ -200,6 +214,23 @@ class AllocationAnalysis {
                    elems * kPtrBitWidth / 8 :
                    elems * srcTy.getElementTypeBitWidth() / 8;
       allocation->addBuffer<BufferT::BufferKind::Scratch>(op, bytes);
+    } else if (auto atomicRMWOp = dyn_cast<triton::AtomicRMWOp>(op)) {
+      auto value = op->getOperand(0);
+      // only scalar requires scratch memory
+      // make it explicit for readability
+      if (value.getType().dyn_cast<RankedTensorType>()) {
+        // nothing to do
```

Comment:
If nothing to do, why keep this branch?

Reply:
make it explicit for readability

Reply:
I personally also feel the empty braces are a little confusing, write the necessary branches I think will more clear considering the existing code style.

Reply:
dyn_cast -> isa ?

Reply:
> dyn_cast -> isa ?
Refer to 
https://github.com/openai/triton/blob/037f9efa95c5bdd86f375183ffa0f1b16a621869/lib/Analysis/Allocation.cpp#L180